core/src/main/java/io/grpc/internal/AbstractServerStream.java

```diff
@@ -31,256 +31,209 @@
 
 package io.grpc.internal;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
 import com.google.common.base.Preconditions;
 
 import io.grpc.Attributes;
 import io.grpc.Metadata;
 import io.grpc.Status;
 
-import java.io.InputStream;
-import java.util.logging.Level;
-import java.util.logging.Logger;
+import javax.annotation.Nullable;
 
 /**
- * Abstract base class for {@link ServerStream} implementations.
- *
- * @param <IdT> the type of the stream identifier
+ * Abstract base class for {@link ServerStream} implementations. Extending classes only need to
+ * implement transportState() and abstractServerStreamSink().
```

Comment:
{@link #transportState}, {@link #abstractServerStreamSink). I like the naming of `abstractServerStreamSink` btw ;-).